nucypher/blockchain/eth/agents.py

```diff
@@ -178,13 +178,27 @@ def increase_allowance(self,
                                                               transacting_power=transacting_power)
         return receipt
 
+    @contract_api(TRANSACTION)
+    def decrease_allowance(self,
+                           transacting_power: TransactingPower,
+                           spender_address: ChecksumAddress,
+                           decrease: NuNits
+                           ) -> TxReceipt:
+        """Decrease the allowance of a spender address funded by a sender address"""
+        contract_function: ContractFunction = self.contract.functions.increaseAllowance(spender_address, decrease)
```

Comment:
```suggestion
        contract_function: ContractFunction = self.contract.functions.decreaseAllowance(spender_address, decrease)
```

Curious how the test didn't catch this...?

Reply:
It wasn't tested, I addressed it in the new commit.